Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1

```diff
@@ -0,0 +1,92 @@
+﻿<#
+ .Synopsis
+  Function for downloading Installed Extensions (symbols) from an online Business Central environment (both AppSource and PTEs)
+ .Description
+  Function for downloading Installed Extensions (symbols) from an online Business Central environment (both AppSource and PTEs)
+ .Parameter bcAuthContext
+  Authorization Context created by New-BcAuthContext. 
+ .Parameter environment
+  Environment from which you want to return the published Apps.
+ .Parameter apiVersion
+  API version. Default is v2.0.
+ .Parameter folder
+  Folder in which the symbols of dependent apps will be placed. Default is $appProjectFolder\symbols.   
+ .Parameter appPublisher
+  Publisher of the app you want to download
+ .Parameter appName
+  Name of the app you want to download
+ .Parameter appVersion
+  Version of the app you want to download
+ .Parameter appId
+  Id of the app you want to download
+ .EXAMPLE
+  $appSymbolsFolder = Join-Path $compilerFolder 'symbols'
+  $authContext = New-BcAuthContext -includeDeviceLogin -tenantID $tenant
+  $bcAuthContext = Renew-BcAuthContext -bcAuthContext $authContext
+  $publishedApps = (Get-BcEnvironmentInstalledExtensions -bcAuthContext $bcAuthContext -environment $environment | ForEach-Object { @{ "Publisher" = $_.Publisher; "Name" = $_.displayName; "Id" = $_.Id; "Version" = [System.Version]::new($_.VersionMajor, $_.VersionMinor, $_.VersionBuild, $_.VersionRevision) } })
+  foreach ($app in $publishedApps) {
+    Download-BcEnvironmentInstalledExtensionToFolder -bcAuthContext $bcAuthContext -environment $environment -appName $app.Name -appId $app.Id -appVersion $app.Version -appPublisher $app.Publisher -folder $appSymbolsFolder
+  }
+  Download-BcEnvironmentInstalledExtensionToFolder -bcAuthContext $bcAuthContext -environment $environment -appName "System" -appVersion "1.0.0.0" -appPublisher "Microsoft" -folder $appSymbolsFolder
+#>
+function Download-BcEnvironmentInstalledExtensionToFolder {
+    param (
+        [Parameter(Mandatory = $true)]
+        [Hashtable] $bcAuthContext,       
+        [string] $environment,
+        [string] $apiVersion = "v2.0",
+        [Parameter(Mandatory = $true)]
+        [alias('appSymbolsFolder')]
+        [string] $folder,
+        [Parameter(Mandatory = $true)]
+        [string] $appPublisher,
+        [Parameter(Mandatory = $true)]
+        [string] $appName,
+        [Parameter(Mandatory = $true)]        
+        [string] $appVersion = '0.0.0.0',
+        [Parameter(Mandatory = $false)]
+        [string] $appId
+    )
+    $telemetryScope = InitTelemetryScope -name $MyInvocation.InvocationName -parameterValues $PSBoundParameters -includeParameters @()
```

Comment:
Normally - I renew the authcontext inside the function.
Reason is that this function might be called deep inside a pipeline, which has been running for more than the validity of the accesstoken.
Then... - remove the renew in the sample above.